packages/completer/src/model.ts

```diff
@@ -459,6 +472,28 @@ export class CompleterModel implements Completer.IModel {
       });
   }
 
+  /**
+   * Lazy load missing data of a completion item by its index.
+   *
+   * @param  activeIndex - the index of item in the completion items list.
+   * @deprecated
+   * @return See `resolveItem` method
+   */
+  private _resolveItemByIndex(
+    activeIndex: number
+  ): Promise<CompletionHandler.ICompletionItem | null> | undefined {
+    if (!this.completionItems) {
+      return undefined;
+    }
+
+    let completionItems = this.completionItems();
```

Comment:
By doing so, we need to change the `Completer.IModel` interface. Do you think we should add a new getter `completionItemsWithIndex` for this?

Reply:
No, I would not go the route of adding a new getter because the state already is hard to follow. An alternative would be a private Map or WeakMap from processed to original item.